v2/tools/generator/internal/jsonast/jsonast.go

```diff
@@ -496,6 +497,13 @@ func getProperties(
 		// add documentation
 		if propSchema.description() != nil {
 			property = property.WithDescription(*propSchema.description())
+		} else if tn, ok := astmodel.AsTypeName(property.PropertyType()); ok {
```

Comment:
I have no issue with this, but another way to do this would be with a stage that ran later and walked the ref and updated the description.

The main downside to making modifications to jsonast is that it feels harder to do complex stuff like "if it's an allOf with a TypeName in it and there's only 1 TypeName, pull that documentation through", whereas that would be more easily doable later with a TypeVisitor/TypeWalker.

Reply:
Done.